exonum-java-binding/core/rust/exonum-java/build.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright 2020 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::env;
+
+fn main() {
+    let lib_paths = [
+        "ROCKSDB_LIB_DIR",
+        "SNAPPY_LIB_DIR",
+        "LZ4_LIB_DIR",
+        "BZ2_LIB_DIR",
+        "Z_LIB_DIR",
+    ];
+
+    lib_paths
+        .iter()
+        .filter_map(|lib| env::var(lib).ok())
+        .for_each(|path| println!("cargo:rustc-search-lib=native={}", path));
+
+    let libs_linking_types = [
```

Comment:
Same here: I'd clarify the intent.

Also, do we link anything statically to the application? Shan't it use these from the `java_bindings` dynamic library?

Reply:
`java_bindings` is dynamic library, right. But it links with its dependencies statically to contain this dependencies inside itself.

Reply:
But this build script is in `exonum-java`, which is not a library, but an executable application. It does not have any effect on the `java_bindings` library, does it?

Reply:
It doesn't. We link statically `librocksdb.a` (and compression libs e.g. `bzip2`, `snappy` etc. which used in rocksdb) to the `exonum-java` application.